include/nlohmann/json.hpp

```diff
@@ -1200,11 +1204,12 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     /// @brief move constructor
     /// @sa https://json.nlohmann.me/api/basic_json/basic_json/
     basic_json(basic_json&& other) noexcept
-        : m_type(std::move(other.m_type)),
+        : json_base_class_t(std::move(other)),
```

Comment:
@barcode Sorry for being so late, but now I got a warning from Clang-Tidy  about accessing `other.m_value` after moving `other` in the line before. Any idea how to fix this?